Makefile.rules

```diff
@@ -155,46 +162,49 @@ SOURCES_ALL += src/compstat_empty.c
 # contain warning flags, and uses absolute include paths.
 ########################################################################
 
+GAP_COMMON_CPPFLAGS =
+
 GAP_CPPFLAGS =
 GAP_PKG_CPPFLAGS =
+GAP_INSTALL_CPPFLAGS =
 
 # First add include paths
 
 # The "build/" directory contains generated header files and thus is
 # placed into the list of include directories.
+# When GAP is installed, this is merged into the $includedir
 GAP_CPPFLAGS += -I$(builddir)/build
 GAP_PKG_CPPFLAGS += -I$(abs_builddir)/build
 
 # The other source files are relative to the srcdir
 GAP_CPPFLAGS += -I$(srcdir)/src
 GAP_PKG_CPPFLAGS += -I$(abs_srcdir)/src
+GAP_INSTALL_CPPFLAGS += -I${includedir}/gap
 
 # For backwards compatibility with certain packages and their kernel
-# extensions, also add the root of the srcdir to the package CPPFLAGS
-GAP_PKG_CPPFLAGS += -I$(abs_srcdir)
+# extensions, also add the root of the srcdir to the package CPPFLAGS. When
+# GAP is installed, this is instead achieved by a symlink named `src` inside
+# ${includedir}/gap which points pack to its parent.
+GAP_PKG_CPPFLAGS += -I$(GAP_LIB_DIR)
 
-# Add GAP_DEFINES (from autoconf, contains -D flags)
-GAP_CPPFLAGS += $(GAP_DEFINES)
-GAP_PKG_CPPFLAGS += $(GAP_DEFINES)
+# Add GAP_DEFINES (from autoconf, contains -D flags) to common flags
+GAP_COMMON_CPPFLAGS += $(GAP_DEFINES)
 
-# Add flags for dependencies
-GAP_CPPFLAGS += $(GMP_CPPFLAGS)
-GAP_CPPFLAGS += $(ZLIB_CPPFLAGS)
-GAP_CPPFLAGS += $(READLINE_CPPFLAGS)
-GAP_CPPFLAGS += $(JULIA_CPPFLAGS)
-GAP_CPPFLAGS += $(BOEHM_GC_CPPFLAGS)
-GAP_CPPFLAGS += $(LIBATOMIC_OPS_CPPFLAGS)
-
-GAP_PKG_CPPFLAGS += $(GMP_CPPFLAGS)
-GAP_PKG_CPPFLAGS += $(ZLIB_CPPFLAGS)
-GAP_PKG_CPPFLAGS += $(READLINE_CPPFLAGS)
-GAP_PKG_CPPFLAGS += $(JULIA_CPPFLAGS)
-GAP_PKG_CPPFLAGS += $(BOEHM_GC_CPPFLAGS)
-GAP_PKG_CPPFLAGS += $(LIBATOMIC_OPS_CPPFLAGS)
+# Add flags for dependencies to common flags
+GAP_COMMON_CPPFLAGS += $(GMP_CPPFLAGS)
+GAP_COMMON_CPPFLAGS += $(ZLIB_CPPFLAGS)
+GAP_COMMON_CPPFLAGS += $(READLINE_CPPFLAGS)
+GAP_COMMON_CPPFLAGS += $(JULIA_CPPFLAGS)
+GAP_COMMON_CPPFLAGS += $(BOEHM_GC_CPPFLAGS)
+GAP_COMMON_CPPFLAGS += $(LIBATOMIC_OPS_CPPFLAGS)
 
-# Finally add user provided flags
-GAP_CPPFLAGS += $(CPPFLAGS)
-GAP_PKG_CPPFLAGS += $(CPPFLAGS)
+# Add user provided flags to common flags
+GAP_COMMON_CPPFLAGS += $(CPPFLAGS)
+
+# Add common flags
+GAP_CPPFLAGS += $(GAP_COMMON_CPPFLAGS)
+GAP_PKG_CPPFLAGS += $(GAP_COMMON_CPPFLAGS)
+GAP_INSTALL_CPPFLAGS += $(GAP_COMMON_CPPFLAGS)
```

Comment:
Just to explain the motivation: we split the CPPFLAGS (and other flags) into three:
- one for use of compiling GAP "in place", i.e. used by `make`
- one (`GAP_PKG_CPPFLAGS`) for putting in `sysinfo.gap` right away, used by packages (this may include e.g. fewer `-I` directives, as packages don't need to be exposed to e.g. the flags for accessing the GNU readline header or GMP etc.)
- one (`GAP_INSTALL_CPPFLAGS`) for the installed version of `sysinfo.gap`, mostly a variant of `GAP_PKG_CPPFLAGS` but with paths adjusted to the installed location.

Also, the same must be done for `GAP_LDFLAGS` and perhaps more

I should probably explain this in a comment.